packages/xrpl/src/utils/collections.ts

```diff
@@ -23,6 +23,7 @@ export function groupBy<T>(
     index,
     arrayReference,
   ) {
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- being safe for js users
     ;(acc[iteratee(value, index, arrayReference)] ||= []).push(value)
```

Comment:
Is it possible to rewrite this line to be more human-readable, or add a comment explaining what it's doing? I'm confused

Reply:
Added to #2406.  I dont know this PR is causing that eslint rule to fail now.  Nothing else changed.